File: jar-shell/src/test/java/com/beancounter/shell/IngestionRequestTest.java
package com.beancounter.shell;

import static org.assertj.core.api.Assertions.assertThat;

import com.beancounter.common.utils.BcJson;
import com.beancounter.shell.ingest.IngestionRequest;
import org.junit.jupiter.api.Test;

class IngestionRequestTest {
  private final BcJson bcJson = new BcJson();

  @Test
  void is_SerializationWorking() throws Exception {

    IngestionRequest ingestionRequest = IngestionRequest.builder()
        .filter("TWEE")
        .portfolioCode("Test")
        .provider("TheProvider")
        .file("123")
        .build();

    assertThat(ingestionRequest.isRatesIgnored()).isTrue();

    String json = bcJson.getObjectMapper().writeValueAsString(ingestionRequest);
    assertThat(bcJson.getObjectMapper().readValue(json, IngestionRequest.class))
        .usingRecursiveComparison().isEqualTo(ingestionRequest);

  }
}

File: jar-shell/src/test/java/com/beancounter/shell/IngestionRequestTest.kt
package com.beancounter.shell

import com.beancounter.common.utils.BcJson
import com.beancounter.shell.ingest.IngestionRequest
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

/**
 * IntegrationRequest Pojo tests
 */
class IngestionRequestTest {
    private val objectMapper = BcJson().objectMapper

    @Test
    @Throws(Exception::class)
    fun is_SerializationWorking() {
        val ingestionRequest = IngestionRequest(
            "",
            file = "TWEE",
            provider = "TheProvider",
            portfolioCode = "Test",
        )
        assertThat(ingestionRequest.ratesIgnored).isTrue
        val json = objectMapper.writeValueAsString(ingestionRequest)
        assertThat(objectMapper.readValue(json, IngestionRequest::class.java))
            .usingRecursiveComparison().isEqualTo(ingestionRequest)
    }
}
